src/extensions/replay/sessionrecording.ts

```diff
@@ -343,13 +361,37 @@ export class SessionRecording {
             return 'buffering'
         }
 
+        if (this.urlTriggerStatus === 'pending') {
+            return 'buffering'
+        }
+
         if (isBoolean(this.isSampled)) {
             return this.isSampled ? 'sampled' : 'disabled'
         } else {
             return 'active'
         }
     }
 
+    private get urlTriggerStatus(): 'activated' | 'pending' | 'disabled' {
+        if (this.receivedDecide && this._urlTriggers.length === 0) {
+            return 'disabled'
+        }
+
+        const currentValue = this.instance?.get_property(SESSION_RECORDING_URL_TRIGGER_ACTIVATED)
+
+        if (TRIGGER_STATUSES.includes(currentValue)) {
+            return currentValue as TriggerStatus
+        }
+
+        return 'pending'
+    }
+
+    private set urlTriggerStatus(status: TriggerStatus) {
+        this.instance?.persistence?.register({
+            [SESSION_RECORDING_URL_TRIGGER_ACTIVATED]: status,
```

Comment:
What would be the best way to evict/expire this value when the recording session times out, so the new recordings have to hit the condition again?

Reply:
we could write the status and the session id to storage to be super explicit...
otherwise there's an onsessionid handler registered elsewhere in this class and i bet it's sufficient to clear the status in that